hikari/impl/rest.py

```diff
@@ -326,17 +330,74 @@ def http_settings(self) -> config.HTTPSettings:
     def proxy_settings(self) -> config.ProxySettings:
         return self._proxy_settings
 
+    @typing.overload
+    def acquire(self, token: typing.Optional[rest_api.TokenStrategy] = None) -> RESTClientImpl:
+        ...
+
+    @typing.overload
     def acquire(
         self,
-        token: typing.Union[str, rest_api.TokenStrategy, None] = None,
+        token: str,
         token_type: typing.Union[str, applications.TokenType] = applications.TokenType.BEARER,
     ) -> RESTClientImpl:
+        ...
+
+    def acquire(
+        self,
+        token: typing.Union[str, rest_api.TokenStrategy, None] = None,
+        token_type: typing.Union[str, applications.TokenType, None] = None,
+    ) -> RESTClientImpl:
+        """Acquire an instance of this REST client.
+
+        !!! note
+            The returned REST client should be started before it can be used,
+            either by calling `RESTClientImpl.start` or by using it as an
+            asynchronous context manager.
+
+        Examples
+        --------
+        ```py
+        rest_app = RESTApp()
+
+        # Using the returned client as a context manager to implicitly start
+        # and stop it.
+        async with rest_app.acquire("A token", "Bot") as client:
+            user = await client.fetch_my_user()
+        ```
+
+        Parameters
+        ----------
+        token : typing.Union[builtins.str, builtins.None, hikari.api.rest.TokenStrategy]
+            The bot or bearer token. If no token is to be used,
+            this can be undefined.
+        token_type : typing.Union[builtins.str, hikari.applications.TokenType, builtins.None]
+            The type of token in use. This should only be passed when `builtins.str`
+            is passed for `token`, can be `"Bot"` or `"Bearer"` and will be
+            defaulted to `"Bearer"` in this situation.
+
+            This should be left as `builtins.None` when either
+            `hikari.api.rest.TokenStrategy` or `builtins.None` is passed for
+            `token`.
+
+        Returns
+        -------
+        RESTClientImpl
+            An instance of the REST client.
+
+        Raises
+        ------
+        ValueError
```

Comment:
```suggestion
        builtins.ValueError
```

Reply:
These do seem to be resolving without prefixing them as `builtins.{0}` though

Reply:
Ik, but we use them everywhere else because some of them dont seem to resolve (I have already tried to fix this several times). It is purely for consistency sake

Reply:
Do these cases where it doesn't resolve even still exist though